stdlib/headernt.c

```diff
@@ -93,7 +93,18 @@ static BOOL WINAPI ctrl_handler(DWORD event)
 #define CP CP_THREAD_ACP
 #endif
 
-#define msg_and_length(msg) msg , (sizeof(msg) - 1)
+static void write_console(HANDLE hOut, WCHAR *wstr)
+{
+  DWORD consoleMode, numwritten, len;
+  static char str[MAX_PATH];
+
+  if (GetConsoleMode(hOut, &consoleMode) != 0) { /* The output stream is a Console */
+    WriteConsoleW(hOut, wstr, wcslen(wstr), &numwritten, NULL);
+  } else { /* The output stream is redirected */
+    len = WideCharToMultiByte(CP, 0, wstr, wcslen(wstr), str, sizeof(str), NULL, NULL);
```

Comment:
It's very unimportant, but I wonder if in principle it wouldn't be better to translate to UTF-8 here (it could be that the displayed filename cannot be represented in the current 8-bit codepage).  Or is the idea of an UTF-8 encoded text file crazy under Windows?

Reply:
Sorry, I missed the fact that CP is indeed UTF-8 when OCaml is configured with Unicode support.

Reply:
Absolutely I think it makes good sense! I just did it this way to preserve existing behavior.